rust/src/types/has_empty.rs

```diff
@@ -0,0 +1,31 @@
+// Copyright (c) 2025 Sonic Operations Ltd
+//
+// Use of this software is governed by the Business Source License included
+// in the LICENSE file and at soniclabs.com/bsl11.
+//
+// Change Date: 2028-4-16
+//
+// On the date above, in accordance with the Business Source License, use of
+// this software will be governed by the GNU Lesser General Public License v3.
+
+/// Traits for types that have an empty variant.
+pub trait HasEmptyNode {
+    /// Returns true if the node is the empty node variant.
+    fn is_empty_node(&self) -> bool;
+
+    /// Creates and returns an empty node.
+    fn make_empty_node() -> Self
```

Comment:
I feel like the `make_*` naming is not very common is Rust.
```suggestion
    fn empty_node() -> Self
```